modules/service/src/main/scala/lucuma/odb/data/ObservationReference.scala

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.data
+
+import cats.Order
+import cats.parse.Parser
+import cats.parse.Parser.*
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.syntax.*
+import lucuma.core.model.ProgramReference
+import lucuma.core.model.parser.ReferenceParsers
+import lucuma.core.optics.Format
+
+case class ObservationReference(
+  programReference: ProgramReference,
+  observationIndex: PosInt
```

Comment:
I would move this to `core` in the future, just as `ProposalReference` and `ProgramReference` started their careers in `lucuma.odb.data` and went on to `lucuma.core.model`.